framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -435,22 +443,24 @@ private AnnotationMirror createBooleanAnnotationMirror(List<Boolean> values) {
 
     private List<Number> calculateNumericalUnaryOp(
             Node operand, NumericalUnaryOps op, TransferInput<CFValue, CFStore> p) {
-        List<? extends Number> lefts = getNumericalValues(operand, p);
         List<Number> resultValues = new ArrayList<>();
-        for (Number left : lefts) {
-            NumberMath<?> nmLeft = NumberMath.getNumberMath(left);
-            switch (op) {
-                case PLUS:
-                    resultValues.add(nmLeft.unaryPlus());
-                    break;
-                case MINUS:
-                    resultValues.add(nmLeft.unaryMinus());
-                    break;
-                case BITWISE_COMPLEMENT:
-                    resultValues.add(nmLeft.bitwiseComplement());
-                    break;
-                default:
-                    throw new UnsupportedOperationException();
+        if (!isIntRange(operand, p)) {
```

Comment:
Ah, thank you for the clarification.
I was not thinking of the distinction.  My main thought was that if you have gone to the trouble to compute something in `Range.java`, it seemed like a shame not to have that capability in the checker.

I think this feature is definitely worth having.  That said, I wouldn't make this feature jeopardize the entire `@IntRange` pull request, if this feature is difficult to implement.